### PR TITLE
Fix assertion when advertise_uri differs

### LINKED ISSUE #972: [1pt] Uri difference during bootstrap results in error
Instance is running localhost:3301.

Request:

```graphql
mutation boot($replicasets: [EditReplicasetInput!]) {
  cluster {
    edit_topology(replicasets: $replicasets) {
      replicasets {
        uuid
      }
    }
  }
}
```

Variables:

```json
{
  "replicasets": [
    {
      "alias": "core1",
      "join_servers": [
        {
          "uri": "127.0.0.1:3301"
        }
      ],
      "roles": []
    }
  ]
}
```

Result:

```json
{
  "errors": [
    {
      "message": "invalid transition Unconfigured -> InitError",
      "extensions": {
        "io.tarantool.errors.stack": "stack traceback:\n\t...rtridge/.rocks/share/tarantool/cartridge/confapplier.lua:114: in function 'set_state'\n\t...rtridge/.rocks/share/tarantool/cartridge/confapplier.lua:323: in function <...rtridge/.rocks/share/tarantool/cartridge/confapplier.lua:280>\n\t[C]: in function 'xpcall'\n\t/opt/cartridge/.rocks/share/tarantool/errors.lua:148: in function </opt/cartridge/.rocks/share/tarantool/errors.lua:132>\n\t...idge/.rocks/share/tarantool/cartridge/remote-control.lua:91: in function <...idge/.rocks/share/tarantool/cartridge/remote-control.lua:64>\n\t[C]: in function 'pcall'\n\t...idge/.rocks/share/tarantool/cartridge/remote-control.lua:264: in function <...idge/.rocks/share/tarantool/cartridge/remote-control.lua:151>\n\t[C]: in function 'xpcall'\n\t/opt/cartridge/.rocks/share/tarantool/errors.lua:148: in function 'pcall'\n\t...idge/.rocks/share/tarantool/cartridge/remote-control.lua:337: in function <...idge/.rocks/share/tarantool/cartridge/remote-control.lua:297>\n\t[C]: in function 'pcall'\n\tbuiltin/socket.lua:1078: in function <builtin/socket.lua:1076>\nduring net.box call to 127.0.0.1:3301, function \"_G.__cartridge_clusterwide_config_commit_2pc\"\nstack traceback:\n\t/opt/cartridge/.rocks/share/tarantool/cartridge/pool.lua:152: in function </opt/cartridge/.rocks/share/tarantool/cartridge/pool.lua:142>",
        "io.tarantool.errors.class_name": "StateError"
      }
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by rosik**
1. Enhance validation during bootstrap
2. Fix invalid state transition
3. Test everything

I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation (unnecessary)

Close #972

### CODE REVIEW

File: cartridge/confapplier.lua [hunk 1]
```diff
@@ -187,6 +187,18 @@ local function validate_config(clusterwide_config, _)
         conf_old = vars.clusterwide_config:get_readonly()
     end
     if conf_old == nil then
+        local instance_uuid = topology.find_server_by_uri(
+            conf_new.topology, vars.advertise_uri
+        )
+        if instance_uuid == nil then
```
Comment (olegrok) [inline cartridge/confapplier.lua:193]: When instance_uuid is not found it's the first case. Is it possible  `instance_uuid  ~= box.info.uuid` (for some reason, e.g. when user makes some manipulations by hand)?

File: cartridge/confapplier.lua [hunk 2]
```diff
@@ -187,6 +187,18 @@ local function validate_config(clusterwide_config, _)
...
+            local err = BootError:new(
+                "Missing %s in clusterwide config," ..
```
Comment (olegrok) [inline cartridge/confapplier.lua:195]: nit: `Missing %s` -> `Missing uri %s` (but feel free to ignore)

File: cartridge/confapplier.lua [hunk 3]
```diff
@@ -362,7 +367,17 @@ local function boot_instance(clusterwide_config)
         log.warn("Will try to rebootsrap, but it may fail again.")
 
         set_state('BootstrappingBox')
-        box_opts.instance_uuid = assert(instance_uuid)
+        if instance_uuid == nil then
+            local err = BootError:new(
+                "Missing %s in clusterwide config." ..
+                " Bootstrap impossible, check advertise_uri correctness",
```
Comment (olegrok) [inline cartridge/confapplier.lua:373]: `Bootstrap impossible` -> `Bootstrap is impossible`?

File: cartridge/lua-api/get-topology.lua [hunk 4]
```diff
@@ -238,7 +238,7 @@ local function get_topology()
                 uri = m.uri,
                 uuid = '',
                 status = 'unconfigured',
-                message = m.payload.error or m.payload.warning or '',
+                message = m.payload.state or '',
```
Comment (olegrok) [inline cartridge/lua-api/get-topology.lua:241]: Don't understand how it's related to current patch. Please describe.

File: test/integration/api_join_test.lua [hunk 5]
```diff
@@ -132,6 +132,20 @@ function g.test_join_server()
         end
     )
 
+    t.assert_error_msg_equals(
+        [[Missing localhost:13302 in clusterwide config,]] ..
```
Comment (olegrok) [inline test/integration/api_join_test.lua:136]: `clusterwide ` -> `cluster-wide `?

File: cartridge/lua-api/get-topology.lua (hunk 4 above)
Comment (rosik) [reply cartridge/lua-api/get-topology.lua:241]: It's necessary for rejoin_test. Both `payload.error` and `payload.warning` are obsolete fields from 1.0. In cartridge v2 the `payload.state` is used instead, but this place wasn't edited in time.

File: cartridge/confapplier.lua (hunk 1 above)
Comment (rosik) [reply cartridge/confapplier.lua:193]: No, it's not. `conf_old == nil` means the instance isn't configured yet. Even manual config editing wouldn't have such effect. Only chopping the code in tiny pieces would.

File: cartridge/lua-api/get-topology.lua (hunk 4 above)
Comment (olegrok) [reply cartridge/lua-api/get-topology.lua:241]: Seems this change should be tested separately

**Review by olegrok — APPROVED** [review]

**Pull request merged**